src/ElasticApm.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\elastic_apm;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+
+use PhilKra\Agent;
+
+/**
+ * The Elastic APM service object.
+ *
+ * @package Drupal\elastic_apm
+ */
+class ElasticApm implements ElasticApmInterface {
+
+  /**
+   * A config object for the elastic_apm configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * The current account.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $account;
+
+  /**
+   * Constructs an Elastic APM object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The config factory service.
+   * @param \Drupal\Core\Session\AccountProxyInterface $account
+   *   The current account.
+   */
+  public function __construct(
+    ConfigFactoryInterface $configFactory,
+    AccountProxyInterface $account
+  ) {
+    $this->config = $configFactory->get('elastic_apm.configuration');
```

Comment:
**CHANGE REQUEST**

*Documentation*

First, the config factory returns `\Drupal\Core\Config\ImmutableConfig` and not `Config` as stated above.

**OPTIONAL CHANGE REQUEST**

*Performance*

Since you're getting the complete data array a few of times  why don't you store it in the property to have the data available? That would improve performance. You're not editing the config object itself in this class.

You can then call the `prepareConfig` function in the constructor (currently `getConfig`) so config is prepared and available anywhere in the class.